kvcoord: include replica info in RangeIterator.Seek into trace

yuzefovich opened the pull request:
This commit modifies the existing "info" log message into an "event"
when seeking the range iterator. This makes it so that the result of the
seek (the replica information) is included into the trace. Additionally,
this commit includes the corresponding message to be included into the
KV trace. The original "info" log message was added about five years ago
and probably hasn't been that useful.

Here is an example of the trace event:
```
key: /NamespaceTable/30/1/100/101/"t"/4/1, desc: r32:/NamespaceTable/{30-Max} [(n1,s1):1, next=2, gen=0]
```

Informs: https://github.com/cockroachlabs/support/issues/1933.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/92947)
<!-- Reviewable:end -->

arulajmani reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> 

> Reviewed 4 of 4 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/92947#-:-NIK-ne19cjREIRKvzAM:bg7igkv)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @AlexTalks)



<!-- Sent from Reviewable.io -->

yuzefovich commented:
TFTR!

bors r+

AlexTalks reviewed (changes requested):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/92947)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @yuzefovich)

___
*[`pkg/kv/kvclient/kvcoord/range_iter.go` line 183 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/92947#-NIKG9KY27FuOFzch4ER:-NIKG9KY27FuOFzch4ES:b-tobb5r) ([raw file](https://github.com/cockroachdb/cockroach/blob/c7c1fd04f8409d694319d838aaedca7657406157/pkg/kv/kvclient/kvcoord/range_iter.go#L183)):*
> ```Go
> func (ri *RangeIterator) Seek(ctx context.Context, key roachpb.RKey, scanDir ScanDirection) {
> 	if log.HasSpanOrEvent(ctx) {
> 		defer func() {
> ```

Question though: why defer this?  once we get to `ri.ds.getRoutingInfo(..)` we are looking up (in a retry loop) the range info in the range cache, which may involve a KV lookup of `meta1`/`meta2` - i.e. a potentially slow RPC. I  I assume it's because you want to incorporate the range info, but wouldn't it be better to potentially do `log.Eventf(..)` both here and after the lookup?


<!-- Sent from Reviewable.io -->

yuzefovich commented:
Let me think about that.

bors r-

yuzefovich committed:
kvcoord: include replica info in RangeIterator.Seek into trace

This commit modifies the existing "info" log message into an "event"
when seeking the range iterator. This makes it so that the result of the
seek (the replica information) is included into the trace. Additionally,
this commit includes the corresponding message to be included into the
KV trace. The original "info" log message was added about five years ago
and probably hasn't been that useful.

Here is an example of the trace event:
```
key: /NamespaceTable/30/1/100/101/"t"/4/1, desc: r32:/NamespaceTable/{30-Max} [(n1,s1):1, next=2, gen=0]
```

Release note: None

yuzefovich reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/92947)* status: :shipit: complete! 0 of 0 LGTMs obtained (and 1 stale) (waiting on @AlexTalks and @arulajmani)

___
*[`pkg/kv/kvclient/kvcoord/range_iter.go` line 183 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/92947#-NIKG9KY27FuOFzch4ER:-NIKPmd5AMFuRtpQ2ekr:b-pgq3vl) ([raw file](https://github.com/cockroachdb/cockroach/blob/c7c1fd04f8409d694319d838aaedca7657406157/pkg/kv/kvclient/kvcoord/range_iter.go#L183)):*
<details><summary><i>Previously, AlexTalks (Alex Sarkesian) wrote…</i></summary><blockquote>

Question though: why defer this?  once we get to `ri.ds.getRoutingInfo(..)` we are looking up (in a retry loop) the range info in the range cache, which may involve a KV lookup of `meta1`/`meta2` - i.e. a potentially slow RPC. I  I assume it's because you want to incorporate the range info, but wouldn't it be better to potentially do `log.Eventf(..)` both here and after the lookup?
</blockquote></details>

I didn't want to pollute the trace too much, but you raise a good point that we might be interested to know how long this `Seek` actually takes. Thus, I changed the code to replace the existing "info" log message into a trace event. That "info" message was introduced in #15413, and probably hasn't been very useful, but please let me know if you think otherwise.


<!-- Sent from Reviewable.io -->

AlexTalks reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/92947#-:-NIZuB8HAtnzHutqUTxn:bnfp4nl)* status: :shipit: complete! 1 of 0 LGTMs obtained (and 1 stale) (waiting on @arulajmani)



<!-- Sent from Reviewable.io -->

yuzefovich commented:
Needed to adjust one logic test file.

TFTRs!

bors r+
